src/App.tsx

```diff
@@ -1,61 +1,133 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleHasError, setTitleHasError] = useState(false);
+  const [userHasError, setUserHasError] = useState(false);
+  const [newTodos, setNewTodos] = useState(todos);
+
+  const addTodo = () => {
+    if (userId && title) {
```

Comment:
please check title if it's not empty in
```suggestion
    if (userId && title.trim()) {
```